appserver/admin/admin-core/src/test/java/com/sun/enterprise/admin/tests/UpgradeTest.java

```diff
@@ -16,63 +17,106 @@
 
 package com.sun.enterprise.admin.tests;
 
-import java.util.logging.Logger;
-import java.util.List;
-import java.util.ArrayList;
-
 import com.sun.enterprise.config.serverbeans.Application;
 import com.sun.enterprise.config.serverbeans.Applications;
+import com.sun.enterprise.config.serverbeans.Config;
 import com.sun.enterprise.config.serverbeans.Domain;
 import com.sun.enterprise.config.serverbeans.Module;
-import com.sun.enterprise.config.serverbeans.Config;
-import org.glassfish.grizzly.config.dom.ThreadPool;
+import com.sun.enterprise.module.bootstrap.StartupContext;
+import com.sun.enterprise.util.SystemPropertyConstants;
+
+import java.util.List;
+import java.util.logging.Logger;
+
+import org.glassfish.api.admin.ServerEnvironment;
 import org.glassfish.api.admin.config.ConfigurationUpgrade;
-import static org.junit.Assert.assertTrue;
-import org.junit.Before;
-import org.junit.Test;
+import org.glassfish.grizzly.config.dom.ThreadPool;
+import org.glassfish.hk2.api.ServiceLocator;
+import org.glassfish.tests.utils.junit.DomainXml;
+import org.glassfish.tests.utils.junit.HK2JUnit5Extension;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import jakarta.inject.Inject;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.arrayContaining;
+import static org.hamcrest.Matchers.hasSize;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
 /**
  * Simple test for the domain.xml upgrade scenario
  *
  * @author Jerome Dochez
  */
-public class UpgradeTest extends ConfigApiTest {
+@ExtendWith(HK2JUnit5Extension.class)
+@DomainXml(value = "UpgradeTest.xml")
+public class UpgradeTest {
+
+    @Inject
+    private ServiceLocator locator;
+    @Inject
+    private StartupContext startupContext;
+
+    // FIXME: Workaround, because ServerEnvironmentImpl changes global System.properties, but other services
```

Comment:
Based on this I created this issue: https://github.com/eclipse-ee4j/glassfish/issues/23593